locals.tf

```diff
@@ -6,6 +6,11 @@ locals {
   create_jump_public_ip = var.create_jump_public_ip == null ? local.is_standard : var.create_jump_public_ip
   create_nfs_public_ip  = var.create_nfs_public_ip == null ? local.is_standard : var.create_nfs_public_ip
 
+  ssh_public_key = ( var.create_jump_vm || var.storage_type == "standard"
+                     ? file(var.ssh_public_key)
+                     : null
+                   )
+                   
```

Comment:
yup, that's what this change is all about - avoid unnecessary checks.
Lazy evaluations will prevent file() to be executed if the condition is not met. 
Otherwise, file() will give a pretty helpful error message.